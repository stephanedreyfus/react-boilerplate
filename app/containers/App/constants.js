/*
 * AppConstants
 * Each action has a corresponding type, which the reducer knows and picks up on.
 * To avoid weird typos between the reducer and the actions, we save them as
 * constants here. We prefix them with 'yourproject/YourComponent' so we avoid
 * reducers accidentally picking up actions they shouldn't.
 *
 * Follow this format:
 * export const YOUR_ACTION_CONSTANT = 'yourproject/YourContainer/YOUR_ACTION_CONSTANT';
 *
 * Constants below for determing action types of adding a phrase and getting phrases.
 */

export const ADD_PHRASE = 'ADD_PHRASE';
export const GET_PHRASES = 'GET_PHRASES';
