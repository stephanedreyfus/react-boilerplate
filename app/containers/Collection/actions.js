/*
 * Collection actions
 */

import {
  GET_PHRASES,
  GET_PHRASES_SUCCESS,
  GET_PHRASES_ERROR,
} from './constants';

export function getPhrases() {
  return {
    type: GET_PHRASES,
  };
}

export function getPhrasesError() {
  return {
    type: GET_PHRASES_ERROR,
  };
}

export function getPhrasesSuccess(phrases) {
  return {
    type: GET_PHRASES_SUCCESS,
    phrases,
  };
}
