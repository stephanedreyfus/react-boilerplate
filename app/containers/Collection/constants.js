export const GET_PHRASES = 'GET_PHRASES';
export const GET_PHRASES_SUCCESS = 'GET_PHRASES_SUCCESS';
export const GET_PHRASES_ERROR = 'GET_PHRASES_ERROR';
