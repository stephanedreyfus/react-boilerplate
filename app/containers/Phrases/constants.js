/*
 * Phrase constants for adding phrases to the database, and
 * viewing all phrases in database.
 */

export const ADD_PHRASE = 'ADD_PHRASE';
export const GET_PHRASES = 'GET_PHRASES';
