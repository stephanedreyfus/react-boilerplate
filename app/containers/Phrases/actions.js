/*
 * Phrase actions
 */

import { ADD_PHRASE } from './constants';

export function addPhrase(phrase) {
  return {
    type: ADD_PHRASE,
    payload: phrase,
  };
}
