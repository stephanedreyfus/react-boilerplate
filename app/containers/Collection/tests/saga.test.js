import { takeLatest, put } from 'redux-saga/effects';

import { getPhrasesError, getPhrasesSuccess } from '../actions';
import getPhraseWatcher, { getPhrases } from '../saga';
import { GET_PHRASES } from '../constants';

describe('getPhrases Saga', () => {
  let getPhrasesGenerator;

  beforeEach(() => {
    getPhrasesGenerator = getPhrases();

    const selectDescriptor = getPhrasesGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();
  });

  it('should dispatch the getPhrasesSuccessfully action if it requests the data successfully', () => {
    const response = [
      {
        phrase: 'test phrase one',
      },
      {
        phrase: 'test phrase two',
      },
    ];
    const putDescriptor = getPhrasesGenerator.next(response).value;
    expect(putDescriptor).toEqual(put(getPhrasesSuccess(response)));
  });

  it('should call the repoLoadingError action if the response errors', () => {
    const response = new Error('Some error');
    const putDescriptor = getPhrasesGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(getPhrasesError(response)));
  });
});

describe('phraseDataSaga watcher', () => {
  const phraseDataSaga = getPhraseWatcher();

  it('should start task to watch for LOAD_REPOS action', () => {
    const takeLatestDescriptor = phraseDataSaga.next().value;
    expect(takeLatestDescriptor).toEqual(takeLatest(GET_PHRASES, getPhrases));
  });
});
