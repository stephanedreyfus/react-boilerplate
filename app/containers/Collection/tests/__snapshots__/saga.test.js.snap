// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPhrases Saga should call the repoLoadingError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "http://localhost:3001/collection",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`getPhrases Saga should dispatch the getPhrasesSuccessfully action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "http://localhost:3001/collection",
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;
